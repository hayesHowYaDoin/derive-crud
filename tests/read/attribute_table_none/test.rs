use derive_crud::Read;

#[derive(Read)]
struct Test {
    #[crud_id]
    id: i64,
    name: String,
    location: String,
}

#[tokio::main]
async fn main() {
    let database_url = std::env::var("DATABASE_URL").unwrap();
    let pool = sqlx::SqlitePool::connect(&database_url).await.unwrap();

    let _ = Test::read(&pool, 1);
}
