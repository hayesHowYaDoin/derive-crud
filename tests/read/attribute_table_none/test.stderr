error: A struct must be annotated with `#[crud_table("table_name")]` to specify the database table
 --> tests/read/attribute_table_none/test.rs:4:1
  |
4 | / struct Test {
5 | |     #[crud_id]
6 | |     id: i64,
7 | |     name: String,
8 | |     location: String,
9 | | }
  | |_^

error[E0599]: no function or associated item named `read` found for struct `Test` in the current scope
  --> tests/read/attribute_table_none/test.rs:13:25
   |
4  | struct Test {
   | ----------- function or associated item `read` not found for this struct
...
13 |     let _result = Test::read(1).await.unwrap(); // Test should fail before unwrap
   |                         ^^^^ function or associated item not found in `Test`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `read`, perhaps you need to implement one of them:
           candidate #1: `AsyncReadExt`
           candidate #2: `std::io::Read`
